addons/OPT/GPS/fn_clientInit.sqf

```diff
@@ -0,0 +1,120 @@
+#include "macros.hpp"
+
+// Inspired by AAW's UnitTracker
+
+GVAR(playerCounter) = 0;
+GVAR(currentIcons) = [];
+GVAR(blockUpdate) = false;
+GVAR(currentHoverGroup) = grpNull;
+GVAR(currentHoverVehicle) = objNull;
+GVAR(groupInfoPFH) = -1;
+GVAR(vehicleInfoPFH) = -1;
+GVAR(lastFrameTriggered) = diag_frameNo;
+
+GVAR(processedIcons) = [];
+GVAR(lastProcessedIcons) = [];
+
+
+// Register Beam dialog map to show unit markers too. Only try doing this once the mission actually started, so everything is hopefully in place.
+// Because we can only do this for an existing display/control, we gotta wait until it actually exists
+["missionStarted", {
+    [{
+        ((findDisplay 444001) displayCtrl 10007) call CFUNC(registerMapControl);
+    }, {!(isNull ((findDisplay 12) displayCtrl 51))}] call CFUNC(waitUntil);
+
+}] call CFUNC(addEventhandler);
+
+
+
+
+
+/*
+The purpose of this stateMachine is to draw mapicons for each unit, group and vehicle whenever a map control is open.
+It draws a single icon per frame to prevent it from hogging resources, while still guaranteeing maximum drawrate.
+Once all icons have been drawn, it cycles back through them in perpetuity. To do this, it has two steps:
+Step 1 initializes _units and _vehicles with what needs to be drawn (init), step 2 draws a single icon (addIcons).
+*/
+GVAR(iconStatemachine) = call CFUNC(createStatemachine);
+
+[GVAR(iconStatemachine), "init", {
+    private _units = allUnits select {[_x] call FUNC(isUnitVisible)};
+
+    // TODO: For now, we ignore empty vehicles
+    // private _vehicles = vehicles select {[_x] call FUNC(isVehicleVisible)};
+    private _vehicles = [];
+
+    // Figure out what icons we had last time. Maybe it's different this time around and we need to remove some icons!
+    GVAR(lastProcessedIcons) = (CGVAR(MapGraphics_MapGraphicsGroup) call CFUNC(allVariables)) select {(_x find toLower QGVAR(IconId)) == 0};
+
+    {
+        DUMP("ICON REMOVED: " + _x);
+        [_x, "hoverin"] call CFUNC(removeMapGraphicsEventHandler);
+        [_x, "hoverout"] call CFUNC(removeMapGraphicsEventHandler);
+        [_x] call CFUNC(removeMapGraphicsGroup);
+    } count (GVAR(lastProcessedIcons) - GVAR(processedIcons));
+
+    // Now we got a clean slate again, let's start the cycle.
+    GVAR(processedIcons) = [];
+
+	if (_units isEqualTo [] && _vehicles isEqualTo []) then {
+        "init"
+    } else {
+        ["addIcons", [_units, _vehicles]]
+    };
+}] call CFUNC(addStatemachineState);
+
+[GVAR(iconStatemachine), "addIcons", {
+    params ["_dummy", "_data"];
+    _data params ["_units", "_vehicles"];
+
+    // First, we process a single Unit
+    if (!(_units isEqualTo [])) then {
+        private _unit = _units deleteAt 0;
+
+        // If it's an invalid unit, move on to the next one
+        while {!([_unit] call FUNC(isUnitVisible)) && {!(_units isEqualTo [])}} do {
```

Comment:
In dieser Schleife sollte auch schon berücksichtigt werden, ob die ausgewählte Einheit überhaupt mit einem Marker versehen wird (vgl. Zeile 82).
Ansonsten verlaufen sich ja etliche Durchgänge der SM ohne dass etwas passiert, da Einheiten verarbeitet werden, die gar nicht gezeichnet werden (dieser Effekt ist denke ich für normale Gruppenmitglieder am größten, da für diese Spieler nur eine Hand voll Marker gemalt werden).